qiskit_quantuminspire/qi_results.py

```diff
@@ -23,21 +23,35 @@ def process(self, job: Job) -> Result:
         """
 
         results = []
+        batch_job_success = [False] * len(self._raw_results)
 
-        for result in self._raw_results:
+        for idx, result in enumerate(self._raw_results):
+            counts = {}
+            shots = 0
+            experiment_success = False
+
+            if isinstance(result, RawJobResult):
+                shots = result.shots_done
+                experiment_success = result.shots_done > 0
+                counts = {hex(int(key, 2)): value for key, value in result.results.items()}
+                batch_job_success[idx] = True
+
+            experiment_data = ExperimentResultData(
+                counts=counts,
+            )
             experiment_result = ExperimentResult(
-                shots=result.shots_done,
-                success=True,
-                data=ExperimentResultData(),
+                shots=shots,
+                success=experiment_success,
+                data=experiment_data,
```

Comment:
How do we associate an ExperimentResult with the job whose result it is? (I suggest you look into `_get_experiment()` of Qiskit's `Result` object).

Reply:
introduced header